HCore-Web/Providers/Impl/DownloadProcessingProxyUrlProviderImpl.cs

```diff
@@ -0,0 +1,197 @@
+            IDataProtectionProvider protectionProvider,
+            IServiceProvider serviceProvider,
+            ILogger<DownloadProcessingProxyUrlProviderImpl> logger
+        )
+        {
+            if (protectionProvider == null)
+            {
+                throw new ArgumentNullException(nameof(protectionProvider));
+            }
+
+            _protector = protectionProvider.CreateProtector(nameof(DownloadProcessingProxyUrlProviderImpl));
+            _serviceProvider = serviceProvider;
+            _logger = logger;
+        }
+
+
+        public virtual Task<Uri> CreateProxyUrlAsync(Uri downloadSourceUri, string fileName, string proxyBaseUrl, string downloadMimeType = null)
+        {
+            if (downloadSourceUri == null)
+            {
+                throw new ArgumentNullException(nameof(downloadSourceUri));
+            }
+
+            if (!downloadSourceUri.IsAbsoluteUri)
+            {
+                throw new ArgumentException("Download source URI is not absolute URI!");
+            }
+
+            if (string.IsNullOrEmpty(proxyBaseUrl))
+            {
+                throw new ArgumentException("No proxy base URL has been provided!");
+            }
+
+            // convert the data to JSON string
+            string uriQueryPayload = ConvertParametersToQueryData(CalculateHashFromParameters(downloadSourceUri, fileName, downloadMimeType));
+
+            UriBuilder proxyUri = new UriBuilder(proxyBaseUrl);
+            if (proxyUri.Query != null && proxyUri.Query.Length > 1)
+            {
+                proxyUri.Query = proxyUri.Query.Substring(1) + "&" + uriQueryPayload;
+            }
+            else
+            {
+                proxyUri.Query = uriQueryPayload;
+            }
+
+            proxyUri.Query = proxyUri.Query.Substring(1) + "&fileName=" + Uri.EscapeDataString(fileName)
+                + (downloadMimeType != null ? "&mime=" + Uri.EscapeDataString(downloadMimeType) : "")
+                + "&u=" + Uri.EscapeDataString(downloadSourceUri.ToString())
+            ;
+            return Task.FromResult(proxyUri.Uri);
+        }
+
+        public virtual async Task<IDownloadFileData> GetFileDataAsync(HttpRequest request, Stream inputData)
+        {
+            Uri downloadUri = new Uri(request.Query["u"]);
+            string fileName = request.Query["fileName"];
+            string mimeType = request.Query["mime"];
+            string characterSet = "utf-8";
+            string originalHash = ConvertParametersFromQueryData(request);
+
+            if (!IsHashValid(originalHash, downloadUri, fileName, mimeType))
+            {
+                throw new UnauthorizedAccessException("Invalid query data");
+            }
+
+            Stream fileData = inputData;
+            if (fileData == null)
+            {
+                var remoteRequest = new HttpRequestMessage(HttpMethod.Get, downloadUri);
+                var client = _serviceProvider.GetService<HttpClient>();
```

Comment:
IHttpClient?

Reply:
Didn't find an interface `IHttpClient` with Riders context sensitive auto-completion. Class `HttpClient` does not implement a usable interface - other than just `IDisposable`.

Do you of such an interface? If so, can you lead me in the proper direction?